applications/GeoMechanicsApplication/custom_utilities/transport_equation_utilities.hpp

```diff
@@ -0,0 +1,65 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Gennady Markelov
+//
+
+#pragma once
+
+// Project includes
+
+// Application includes
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class GeoTransportEquationUtilities
+{
+public:
+    using VectorType = Vector;
+    using MatrixType = Matrix;
+
+    template <unsigned int TDim, unsigned int TNumNodes>
+    static inline void CalculatePermeabilityMatrixH(BoundedMatrix<double, TNumNodes, TNumNodes>& rPMatrix,
+                                                    const Matrix& GradNpT,
+                                                    double        DynamicViscosityInverse,
+                                                    const BoundedMatrix<double, TDim, TDim>& PermeabilityMatrix,
```

Comment:
Perhaps we should discuss with @WPK4FEM about a better name for this function parameter. The function is named `CalculatePermeabilityMatrix`, but one of its parameters (not being the output parameter) is called `PermeabilityMatrix`. This is rather confusing to me.

Reply:
Wijtze-Pieter agreed to call it as MaterialPermeabilityMatrix.